common/src/main/java/com/wynntils/commands/CompassCommand.java

```diff
@@ -0,0 +1,177 @@
+import com.wynntils.core.commands.CommandBase;
+import com.wynntils.mc.objects.Location;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.utils.StringUtils;
+import com.wynntils.wynn.model.CompassModel;
+import com.wynntils.wynn.model.map.MapModel;
+import com.wynntils.wynn.model.map.poi.LabelPoi;
+import com.wynntils.wynn.model.map.poi.Poi;
+import com.wynntils.wynn.model.map.poi.ServiceKind;
+import com.wynntils.wynn.model.map.poi.ServicePoi;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import net.minecraft.ChatFormatting;
+import net.minecraft.commands.CommandSourceStack;
+import net.minecraft.commands.Commands;
+import net.minecraft.commands.arguments.coordinates.Coordinates;
+import net.minecraft.commands.arguments.coordinates.Vec3Argument;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.world.phys.Vec3;
+
+public class CompassCommand extends CommandBase {
+    @Override
+    public LiteralArgumentBuilder<CommandSourceStack> getBaseCommandBuilder() {
+        return Commands.literal("compass")
+                .then(Commands.literal("at")
+                        .then(Commands.argument("location", Vec3Argument.vec3()).executes(this::compassAt))
+                        .build())
+                .then(Commands.literal("service")
+                        .then(Commands.argument("name", StringArgumentType.greedyString())
+                                .suggests(LocateCommand.SERVICE_SUGGESTION_PROVIDER)
+                                .executes(this::compassService))
+                        .build())
+                .then(Commands.literal("place")
+                        .then(Commands.argument("name", StringArgumentType.greedyString())
+                                .suggests(LocateCommand.PLACES_SUGGESTION_PROVIDER)
+                                .executes(this::compassPlace))
+                        .build())
+                .then(Commands.literal("npc")
+                        .then(Commands.argument("name", StringArgumentType.greedyString())
+                                .executes(this::notImplemented))
+                        .build())
+                .then(Commands.literal("other")
+                        .then(Commands.argument("name", StringArgumentType.greedyString())
+                                .executes(this::notImplemented))
+                        .build())
+                .executes(this::syntaxError);
+    }
+
+    private int compassAt(CommandContext<CommandSourceStack> context) {
+        Coordinates coordinates = Vec3Argument.getCoordinates(context, "location");
+        Location location = new Location(coordinates.getBlockPos(context.getSource()));
+        CompassModel.setCompassLocation(location);
+
+        MutableComponent response = new TextComponent("Setting compass to ").withStyle(ChatFormatting.AQUA);
```

Comment:
maybe "Set compass to x y z" is better, as it isnt really a continuous process

Reply:
That sounds like an imperative to the user... Maybe "Compass set to ..."?

Reply:
Yeah